tests/frontend/test_parameters.py

```diff
@@ -206,3 +225,127 @@ def test_parameter_unary_negation(self, p):
         pp.val = p
         assert par_evaluate(-p) == pytest.approx(-p)
         assert par_evaluate(-pp) == pytest.approx(-p)
+
+
+def mock_run_prog(self, prog, **kwargs):
+    """Mock function that uses a global variable for constructing
+    the engine queue when running programs."""
+    global applied
+    for cmd in prog.circuit:
+        try:
+            cmd.op.apply(cmd.reg, self.backend, **kwargs)
+            applied.append(cmd)
+        except NotImplementedError:
+            for c in cmd.op._decompose(cmd.reg, **kwargs):
+                c.op.apply(c.reg, self.backend, **kwargs)
+                applied.append(c)
+    return applied
+
+
+class TestParameterTFIntegration:
+    """Test integration of the parameter handling system with
+    various gates and TensorFlow."""
+    pytest.importorskip("tensorflow", minversion="2.0")
+
+    @pytest.mark.parametrize("gate", [sf.ops.Dgate, sf.ops.Sgate, sf.ops.Coherent])
+    def test_single_mode_gate_complex_phase(self, backend, gate, monkeypatch):
+        """Test single mode gates with complex phase arguments"""
+        import tensorflow as tf
+
+        prog = sf.Program(1)
+
+        # create symbolic parameters
+        r = prog.params('r')
+        phi = prog.params('phi')
+
+        with prog as q:
+            gate(r, phi) | q[0]
+
+        # create numeric values
+        mapping = {'r': tf.Variable(0.1), 'phi': tf.Variable(0.1)}
+        prog.bind_params(mapping)
+
+        # verify bound parameters are correct
+        assert prog.free_params['r'].val is mapping['r']
+        assert prog.free_params['phi'].val is mapping['phi']
+
+        # assert executed program is constructed correctly
+        global applied
+        cmds = []
+        applied = []
+
+        with monkeypatch.context() as m:
+            m.setattr(sf.LocalEngine, "_run_program", mock_run_prog)
+            eng = sf.LocalEngine(backend)
+            result = eng.run(prog, args=mapping)
+
+        assert isinstance(applied[0].op, gate)
```

Comment:
Just to make sure that there are no leftover operations which were not tested for (this is reoccurring in the other test cases as well)
```suggestion
        assert isinstance(applied[0].op, gate)
        assert len(applied) == 1
```